source/FreeRTOS_Routing.c

```diff
@@ -869,13 +869,13 @@ struct xIPv6_Couple
                                                    pcBufferFrom,
                                                    sizeof( pcBufferFrom ) );
 
-                FreeRTOS_debug_printf( ( "EasyFit[%x]: %d %d %d ( %s ->%s ) BAD\n",
-                                         usFrameType,
-                                         ( unsigned ) xCount[ 0 ],
-                                         ( unsigned ) xCount[ 1 ],
-                                         ( unsigned ) xCount[ 2 ],
-                                         ( xRetNtopFrom == NULL ) ? "INVALID" : pcBufferFrom,
-                                         ( xRetNtopTo == NULL ) ? "INVALID" : pcBufferTo ) );
+                FreeRTOS_printf( ( "EasyFit[%x]: %d %d %d ( %s ->%s ) BAD\n",
```

Comment:
In an earlier PR, I don't remember when, `FreeRTOS_printf()` was replaced with `FreeRTOS_debug_printf()`, because the messages weren't interesting enough.

Now in stead of reverting this PR, I would propose to replace `ipconfigHAS_PRINTF` with `ipconfigHAS_DEBUG_PRINTF`:

~~~diff
-        #if ( ipconfigHAS_PRINTF != 0 )
+        #if( ipconfigHAS_DEBUG_PRINTF != 0 )
             if( pxReturn == NULL )
             {
                 FreeRTOS_debug_printf( );
             }
~~~

Reply:
I looked it up: in PR #855 , the message was still printed by calling `FreeRTOS_printf()`. Later it was changed to `FreeRTOS_debug_printf()`, without adapting `ipconfigHAS_PRINTF`.